go/test/endtoend/sharding/verticalsplit/vertical_split_test.go

```diff
@@ -0,0 +1,766 @@
+
+	// RebuildKeyspaceGraph source keyspace
+	_ = clusterInstance.VtctlclientProcess.ExecuteCommand("RebuildKeyspaceGraph", sourceKeyspace)
+
+	// RebuildKeyspaceGraph destination keyspace
+	_ = clusterInstance.VtctlclientProcess.ExecuteCommand("RebuildKeyspaceGraph", destinationKeyspace)
+
+	// source schema
+	for _, tablet := range []cluster.Vttablet{sourceMasterTablet, sourceReplicaTablet, sourceRdOnlyTablet1, sourceRdOnlyTablet2} {
+		for _, tableName := range tableArr {
+			_, err := tablet.VttabletProcess.QueryTablet(fmt.Sprintf(createTabletTemplate, tableName), sourceKeyspace, true)
+			assert.Nil(t, err)
+		}
+		_, err := tablet.VttabletProcess.QueryTablet(fmt.Sprintf(createViewTemplate, "view1", "moving1"), sourceKeyspace, true)
+		assert.Nil(t, err)
+
+		// RBR only because Vitess requires the primary key for query rewrites if
+		// it is running with statement based replication.
+		_, err = tablet.VttabletProcess.QueryTablet(createMoving3NoPkTable, sourceKeyspace, true)
+		assert.Nil(t, err)
+	}
+
+	// destination schema
+	// Insert data directly because vtgate would redirect us.
+	for _, tablet := range []cluster.Vttablet{destinationMasterTablet, destinationReplicaTablet, destinationRdOnlyTablet1, destinationRdOnlyTablet2} {
+		_, err = tablet.VttabletProcess.QueryTablet(fmt.Sprintf(createTabletTemplate, "extra1"), destinationKeyspace, true)
+		assert.Nil(t, err)
+	}
+
+	// source and destination master and replica tablets will be started
+	for _, tablet := range []cluster.Vttablet{sourceMasterTablet, sourceReplicaTablet, destinationMasterTablet, destinationReplicaTablet} {
+		_ = tablet.VttabletProcess.Setup()
+	}
+
+	// rdonly tablets will be started
+	for _, tablet := range []cluster.Vttablet{sourceRdOnlyTablet1, sourceRdOnlyTablet2, destinationRdOnlyTablet1, destinationRdOnlyTablet2} {
+		_ = tablet.VttabletProcess.Setup()
+	}
+
+	// check SrvKeyspace
+	ksServedFrom := "ServedFrom(master): source_keyspace\nServedFrom(rdonly): source_keyspace\nServedFrom(replica): source_keyspace\n"
+	checkSrvKeyspaceServedFrom(t, cellj, destinationKeyspace, ksServedFrom, *clusterInstance)
+
+	// reparent to make the tablets work (we use health check, fix their types)
+	err = clusterInstance.VtctlclientProcess.InitShardMaster(sourceKeyspace, shardName, cellj, sourceMasterTablet.TabletUID)
+	assert.Nil(t, err)
+	err = clusterInstance.VtctlclientProcess.InitShardMaster(destinationKeyspace, shardName, cellj, destinationMasterTablet.TabletUID)
+	assert.Nil(t, err)
+
+	sourceMasterTablet.Type = "master"
+	destinationMasterTablet.Type = "master"
+
+	for _, tablet := range []cluster.Vttablet{sourceReplicaTablet, destinationReplicaTablet} {
+		_ = tablet.VttabletProcess.WaitForTabletType("SERVING")
+		assert.Nil(t, err)
+	}
+	for _, tablet := range []cluster.Vttablet{sourceRdOnlyTablet1, sourceRdOnlyTablet2, destinationRdOnlyTablet1, destinationRdOnlyTablet2} {
+		_ = tablet.VttabletProcess.WaitForTabletType("SERVING")
+		assert.Nil(t, err)
+	}
```

Comment:
We can combine the above 2 loops

Reply:
It is as per the python testcase which is to wait first for the master and replicas and then for Rdonly's.

Reply:
ok